### PR TITLE
(SCT-1310) feature - POST end point for adding case status answer for LAC

### PR DESCRIPTION
**Pull request by LBHTKarki**
## Link to JIRA ticket

https://hackney.atlassian.net/browse/SCT-1310

### *What is the problem we're trying to solve*

We need an endpoint that let us record changes to child's circumstances

### *What changes have we introduced*

Add new POST end point that allows us to add new answers to the case status which are used to indicate changes in circumstances.

Only LAC status can have these additional answers, so requests to add answers to other types will be rejected.

Also tidied up the unrelated CreateCaseStatusRequestTest.cs file.

#### _Checklist_

- [ ] Added end-to-end (i.e. integration) tests where necessary e.g to test complete functionality of a newly added feature
- [x] Added tests to cover all new production code
- [ ] Added comments to the README or updated relevant documentation _(add link to documentation)_, where necessary.
- [x] Checked all code for possible refactoring
- [x] Code pipeline builds correctly

### CODE REVIEW

File: SocialCareCaseViewerApi/V1/Boundary/Requests/CreateCaseStatusAnswerRequest.cs [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+            RuleFor(pr => pr.Answers).Must(x => x.Count > 0).WithMessage("Answers must be provided");
+
+            RuleForEach(pr => pr.Answers)
+                .ChildRules(field =>
+                {
+                    field.RuleFor(t => t.Option).NotEmpty().WithMessage("Option must not be empty");
+                    field.RuleFor(t => t.Value).NotEmpty().WithMessage("Value must not be empty");
+                });
+
+            RuleFor(x => x.StartDate)
```
Comment (mapleleafjack) [inline SocialCareCaseViewerApi/V1/Boundary/Requests/CreateCaseStatusAnswerRequest.cs:40]: I think Case Status Answers should be both in the future and in the past, as we can have future pending status changes

File: SocialCareCaseViewerApi.Tests/V1/Boundary/Request/CreateCaseStatusAnswersRequestTests.cs [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+            var response = _createCaseStatusAnswerRequestValidator.Validate(requestWithoutAnswers);
+
+            response.IsValid.Should().BeFalse();
+            response.Errors.Should().Contain(e => e.ErrorMessage == "Option must not be empty");
+            response.Errors.Should().Contain(e => e.ErrorMessage == "Value must not be empty");
+        }
+
+        [Test]
+        public void WhenStartDateIsInTheFutureReturnsErrorWithMessage()
```
Comment (mapleleafjack) [inline SocialCareCaseViewerApi.Tests/V1/Boundary/Request/CreateCaseStatusAnswersRequestTests.cs:54]: I think Case Status Answers should be both in the future and in the past, as we can have future pending status changes

File: SocialCareCaseViewerApi/V1/Controllers/CaseStatusController.cs [hunk 3]
```diff
@@ -81,6 +81,42 @@ public IActionResult CreateCaseStatus([FromBody] CreateCaseStatusRequest request
...
+                return CreatedAtAction(nameof(CreateCaseStatusAnswers), "Successfully created case status answers.");
+            }
+            catch (Exception e) when (
+                e is WorkerNotFoundException)
+            {
+                return BadRequest(e.Message);
+            }
+            catch (CaseStatusDoesNotExistException e)
+            {
+                return NotFound(e.Message);
```
Comment (JohnFarrellDev) [inline SocialCareCaseViewerApi/V1/Controllers/CaseStatusController.cs:116]: My only suggestion is that I don't think a POST endpoint should be returning a 404, if the Case Status does not exist then the request was bad and a 400 seems appropriate.

File: SocialCareCaseViewerApi/V1/Boundary/Requests/CreateCaseStatusAnswerRequest.cs (hunk 1 above)
Comment (LBHTKarki) [reply SocialCareCaseViewerApi/V1/Boundary/Requests/CreateCaseStatusAnswerRequest.cs:40]: @mapleleafjack  That's true. I've pushed an update to allow both future and past dates

**Pull request merged**